src/StorageFolding.cpp

```diff
@@ -103,7 +103,10 @@ class FoldStorageOfFunction : public IRMutator {
                 Expr new_mins = Call::make(type_of<int *>(), Call::make_struct, mins, Call::Intrinsic);
                 vector<Expr> new_args = op->args;
                 new_args[3] = new_mins;
-                expr = Call::make(op->type, op->name, new_args, op->call_type);
+                internal_assert(op->type == type_of<halide_buffer_t *>() &&
```

Comment:
This mutates an existing buffer_crop call, so isn't this going to redundantly wrap it in another require?

Reply:
hmm, yes, I suppose it could -- let me see if it will get optimized away or not

Reply:
The gotcha here is that this code transmutes `_halide_buffer_crop()` -> `_halide_buffer_set_bounds(_halide_buffer_crop())`; if we leave out the `require()`, we can risk passing nullptr to the enclosing call. I suppose we could just make that call check for an incoming nullptr and bail early...